### PR TITLE
[FIX] sms: various typos

### PR DESCRIPTION
**Pull request by rrahir**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sms/models/sms_sms.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def _send(self, delete_all=False, raise_exception=False):
     def _postprocess_sent_sms(self, iap_results, failure_reason=None, delete_all=False):
         sms_to_notif_status = {
             'success': False, 'insufficient_credit': 'sms_credit',
-            'wrong_format_number': 'sms_number_format', 'server_error': 'sms_server'}
+            'wrong_number_format': 'sms_number_format', 'server_error': 'sms_server'}
```
Comment (tde-banana-odoo) [inline addons/sms/models/sms_sms.py]: Hello,

https://github.com/odoo/iap-apps/blob/master/iap_services/iap_proxy_sms/controllers/main.py tells the inverse and that wrong_format_number is returned.

File: addons/sms/models/mail_thread.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def _compute_message_has_sms_error(self):
 
     @api.model
     def _search_message_has_sms_error(self, operator, operand):
-        return ['&', ('message_ids.has_sms_error', operator, operand), ('message_ids.author_id', '=', self.env.user.partner_id.id)]
+        return ['&', ('message_ids.has_sms_error', operator, operand), ('message_ids.author_id.id', '=', self.env.user.partner_id.id)]
```
Comment (tde-banana-odoo) [inline addons/sms/models/mail_thread.py]: You normally don't have to put .id in a domain about m2o. What was the issue ?

Comment (rrahir) [reply addons/sms/models/mail_thread.py]: Yup, it was a wip that I forgot to remove, I'll delete this from the PR

File: addons/sms/models/sms_sms.py (hunk 1 above)
Comment (rrahir) [reply addons/sms/models/sms_sms.py]: There is and inconsistency between the doc of the route and the actual result returned. 
e.g.

https://github.com/odoo/iap-apps/blob/master/iap_services/iap_proxy_sms/controllers/main.py#L125
https://github.com/odoo/iap-apps/blob/master/iap_services/iap_proxy_sms/controllers/main.py#L162

We could make the change server-side but I think that 'wrong_number_format' suits better.

Comment (tde-banana-odoo) [reply addons/sms/models/sms_sms.py]: I propose that you update the doc then :D .

Comment (tde-banana-odoo) [reply addons/sms/models/sms_sms.py]: I will propagate wrong_number_format everywhere in code of odoo.

Comment (rrahir) [reply addons/sms/models/sms_sms.py]: on it :p

**Comment by rrahir** [comment]
changes added in PR : https://github.com/odoo/odoo/pull/35025

**Pull request closed without merging**